src/components/PageDefinitions.js

```diff
@@ -199,11 +199,11 @@ class PageDefinitions extends AbstractPageDefinitions {
     })
   }
 
-  onAddComponent(value, after = null) {
+  async onAddComponent(value, after = null) {
     const { dispatch, token, definitions } = this.props
     const component = typeof value === 'string' ? EntitySpec.fromPath(value) : value
     const path = component.toPath()
-    !definitions.entries[path] && dispatch(getDefinitionsAction(token, [path]))
+    !definitions.entries[path] && (await dispatch(getDefinitionsAction(token, [path])))
```

Comment:
That's correct. I added this because otherwise the list would be updated with the component and if getting the definitions takes a long time, the list Attributions would be empty because a react component would have already rendered.

This component, for example, takes 7/9 seconds to return it's definition
http://localhost:3000/definitions/git/github/microsoft/vscode/490ef761b76b3f3b3832eff7a588aac891e5fe80

Reply:
That is why we have sequence numbers etc.  The details of the components fill in as they are added to the list. We should prevent opening any details etc until the component is loaded. I could be wrong but it does seem wrong to "block" the UI event on IO. 

As for why it takes a while to get the definition, that is interesting. I would have expected that to be much faster since it is essentially a direct blob access. I wonder if we have regressed somewhere. (separate problem but something to watch)

Reply:
I reverted these changes and the react component rendering issue no longer occurs